app/controllers/admin/categories_controller.rb

```diff
@@ -0,0 +1,57 @@
+class Admin::CategoriesController < Admin::AdminController
+  before_action :load_category, only: %i(edit destroy update)
+
+  def index
+    @pagy, @category = pagy Category.all, items: Settings.item_per_page
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new category_params
+    if @category.save
+      flash[:success] = t "success"
+      redirect_to admin_categories_url
+    else
+      flash[:danger] = t "create_fail"
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    if @category.update category_params
+      flash[:success] = t "success"
+      redirect_to admin_categories_url
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    if @category.products.present?
```

Comment:
Nhuư này mới chuẩn em nhé
chớ ko nó vẫn chạy xuống check điều kiện dưới
```suggestion
    if @category.products.present?
      flash[:warning] = t "product_still_have"
    else
      if @category.destroy
        flash[:success] = t "success"
      else
        flash[:danger] = t "fails"
      end
    end
    redirect_to admin_categories_path
```